### ISSUE TITLE
Introduce fisher list and deprecate fisher --list

### ISSUE THREAD

**Issue by bucaran** [fisherman/fisherman#134]
Use `fisher list` instead of `fisher --list`. The behavior was modified only slightly, but everything should work as usual.

![lvss](https://cloud.githubusercontent.com/assets/8317250/13388184/43c998e2-df01-11e5-9464-4fdece4d65a2.gif)

If you are confused about the difference between _list_ and _search_, here is the gist of it. List queries the local system ($fisher_cache) and search queries the remote index ($fisher_index).

 A slightly more complicated explanation is, list, shows you what you have downloaded to $fisher_cache, and uses "decorators", `*`, `>` and `@` to indicate whether the item is a plugin, a prompt or a symbolic link. If an item does not have a decorator, then it's downloaded to the cache, but it's **not** activated. To activated it use `fisher install <plugin name>` and to disable it again use `fisher uninstall <plugin name>`. If you want _it_ gone gone, `fisher uninstall <plugin name> --force`. List is fast.

Search, on the other hand, shows you the contents of the [index](https://github.com/fisherman/fisher-index/blob/master/index), which is the "database" where all the official plugins are registered. Search is also smart enough to show you which of the plugins from the list are installed or are prompts. Search is slightly slower than list and it may update the index automatically based in the state of the internal `$fisher_last_update` time variable.

So, in short:

```fish
fisher list # show me what's installed
```

```fish
fisher search # show me what's available
```
